### PR TITLE
Netn nov2019 updates

### PR DESCRIPTION
**Pull request by KateMMiller**
- Tweaked compileNETNdata to better match the data that feed into the package, fixed/simplified code to pull out the MDL/UDL values, added error handling, and made saving options more flexible. 
- Added MDL and UDL to import function
- Added nonparTrends function that calculates non-parametric trends with water data. Non-censored data are analyzed with the TheilSen and deseasoned=T. Censored data were analyzed using cenken function mapped to each month in the dataset.

### CODE REVIEW

**Comment by KateMMiller** [comment]
Just noticed while working on the visualizer that pH measurements in streams outside of ACAD aren't being included in the compileNETNdata(). The issue is in the code below. In ACAD, sample depth is text, but in other parks, sample depth is a number.

`if(surface == TRUE){
  #select only 'stream', 'epilimnion', NA Sample Depths, or depths <2m.
  waterDat <- waterDat %>%
    filter(SampleDepth == 'stream' | SampleDepth == 'epilimnion' | is.na(SampleDepth) )
} `

I changed this to (is <2m an appropriate cutoff?): 
`if(surface == TRUE){
  #select only 'stream', 'epilimnion', NA Sample Depths, or depths <2m.
  waterDat <- waterDat %>%
    filter(SampleDepth == 'stream' | SampleDepth == 'epilimnion' | is.na(SampleDepth) | 
            SampleDepth <2)
} `

If that's not a correct cutoff, then suggestion a different one (or I can just allow any numeric depth).

**Comment by KateMMiller** [comment]
I added Censored and ValueCen to the Water Data.csv, so I can use those fields to direct plotting in the visualizer. I added a check for these fields in the importNCRNWater function, and added it to the Data slot (like MDL and UDL). I initially made this reactive in the water visualizer (eg check if metric is censored for a site, and if it is, create and plot ValueCen), but it slows things down. I decided to just include these with the data and import so speed is less compromised in the viz. This should be the last change I make to existing functions in NCRNWater. I may add a different plotting function for censored data, but only after I make a few tweaks to the NCRNWaterViz first.

File: R/compileNETNdata.R [hunk 1]
```diff
@@ -0,0 +1,651 @@
+#' @title compileNETNdata
+#' 
+#' @description Compile flat files from NETN Access Database for R package 
+#' 
+#' @section Warning:
```
Comment (johnpauls) [inline R/compileNETNdata.R:5]: What is the "@section Warning:" for

**Comment by KateMMiller** [comment]
OK, I think I'm finished working on NCRNWater for awhile. The latest batch of commits was mostly to make the output from nonparTrends more compatible with the visualizer, and to make it possible to plot censored data and split the data into months in the waterseries plot.

**Comment by aaronweed** [comment]
Wow Kate! Thank you for working on this! I am checking the compiler function now and works great for me. Bill suggested, and we agreed, that summarizing the lake/pond data within the top 2m would be an appropriate cutoff to represent the epilimnion. Of course the actual epilimnion depth may be able to be estimated from the data but for a first crack Bill felt 2m was a good cutoff. 

The only minor changes I made to `compileNETN` so far were adding the `SampleDepth` and `Depth Units` fields to the WaterData.csv export and I fixed a minor typo in the Roxygen header. I added these fields because I think they will be helpful should the data be exported ever via the viz and when integrating into the viz.  For instance, I noticed in the viz that the time series plot of lake data (e.g. Eagle, temp) shows data across all depths. The purpose of the `SampleDepth` column was to be used partly to filter the dataset to only plot the time series of surface/epilimnion values. 

JP- would you ever need to use sample depth in the package for your data (or do you collect it)? I can easily add these two cols in the `importNCRNWater` func (line 51 I think), or do you just want us to create a NETN specific import function, or can we just remove the args to `select` on line 51 so all data fields are passed to the @Data slot (e.g. `waterobj[[1]]@Sites[[1]]@Characteristics[[1]][[22]]@Data`) ? Of course if I add those fields your data would need them as well. 
 
I will commit the changes I mentioned above to `compileNETNdata` now and continue checking the other functions. 

`nonparTrends` seems to be working fine as well but I still need to check further. I think `getCharInfo` should be added to @include (but seems to be working without it) and what do you think about being able to access additional arguments (via `...`) to the `TheilSen` rather than `getWData` function (or both)? Currently I don't think `nonparTrends` allows you to access additional args from either `TheilSen` or `getWData` based on a quick view and test of the code. The `...` need to be added to one of the functions. (This is an area I am still trying to learn how it works!)

Great work and thanks again for thinking through this. I am excited to review the viz changes too! I will get to that ASAP and finish up reviewing this soon as well.

**Comment by aaronweed** [comment]
Good morning! So I also committed a few more small updates to `compileNETNdata` and `importNCRNwater` for more flexible handling of column names due to network differences. For instance, Kate you had `importNCRNwater` mutating cols that might not be in JP's data so I just added an `if(any(name())` statement to handle when `ValueCen` and `Censored` exist in `Indata`. I also updated `importNCRNWater` to be flexible when subsetting columns into the Data slot so that `SampleDepth` and `Depth Units` (or other network-specific field names) passes to the water data object (for NETN) but if you don't have these cols the function won't fail. I checked your data JP and was able to import successfully. Do you anticipate these changes causing any other downstream issues? (excuse the pun!)

**Comment by KateMMiller** [comment]
Thanks for looking this over and for the suggestions Aaron. I added the ... to the function arguments and the getWData line, and tested it using a range of years. It seems to be working for me. All the other changes you made sound good to me. I think it's ready for JP's review.

**Review by johnpauls — COMMENTED** [review]
minor changes pushed to branch

**Review by johnpauls — APPROVED** [review]

**Pull request merged**